File: csharp/src/main/CsharpDotNet2/spoonacular/Model/InlineResponse20030.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace spoonacular.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse20030 {
    /// <summary>
    /// Gets or Sets Id
    /// </summary>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// Gets or Sets Title
    /// </summary>
    [DataMember(Name="title", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "title")]
    public string Title { get; set; }

    /// <summary>
    /// Gets or Sets Breadcrumbs
    /// </summary>
    [DataMember(Name="breadcrumbs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "breadcrumbs")]
    public List<string> Breadcrumbs { get; set; }

    /// <summary>
    /// Gets or Sets ImageType
    /// </summary>
    [DataMember(Name="imageType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "imageType")]
    public string ImageType { get; set; }

    /// <summary>
    /// Gets or Sets Badges
    /// </summary>
    [DataMember(Name="badges", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "badges")]
    public List<string> Badges { get; set; }

    /// <summary>
    /// Gets or Sets ImportantBadges
    /// </summary>
    [DataMember(Name="importantBadges", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "importantBadges")]
    public List<string> ImportantBadges { get; set; }

    /// <summary>
    /// Gets or Sets IngredientCount
    /// </summary>
    [DataMember(Name="ingredientCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ingredientCount")]
    public int? IngredientCount { get; set; }

    /// <summary>
    /// Gets or Sets GeneratedText
    /// </summary>
    [DataMember(Name="generatedText", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "generatedText")]
    public Object GeneratedText { get; set; }

    /// <summary>
    /// Gets or Sets IngredientList
    /// </summary>
    [DataMember(Name="ingredientList", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ingredientList")]
    public string IngredientList { get; set; }

    /// <summary>
    /// Gets or Sets Ingredients
    /// </summary>
    [DataMember(Name="ingredients", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ingredients")]
    public List<InlineResponse20030Ingredients> Ingredients { get; set; }

    /// <summary>
    /// Gets or Sets Likes
    /// </summary>
    [DataMember(Name="likes", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "likes")]
    public decimal? Likes { get; set; }

    /// <summary>
    /// Gets or Sets Aisle
    /// </summary>
    [DataMember(Name="aisle", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "aisle")]
    public string Aisle { get; set; }

    /// <summary>
    /// Gets or Sets Nutrition
    /// </summary>
    [DataMember(Name="nutrition", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "nutrition")]
    public InlineResponse20028Nutrition Nutrition { get; set; }

    /// <summary>
    /// Gets or Sets Price
    /// </summary>
    [DataMember(Name="price", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "price")]
    public decimal? Price { get; set; }

    /// <summary>
    /// Gets or Sets Servings
    /// </summary>
    [DataMember(Name="servings", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "servings")]
    public InlineResponse20028Servings Servings { get; set; }

    /// <summary>
    /// Gets or Sets SpoonacularScore
    /// </summary>
    [DataMember(Name="spoonacularScore", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "spoonacularScore")]
    public decimal? SpoonacularScore { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse20030 {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Title: ").Append(Title).Append("\n");
      sb.Append("  Breadcrumbs: ").Append(Breadcrumbs).Append("\n");
      sb.Append("  ImageType: ").Append(ImageType).Append("\n");
      sb.Append("  Badges: ").Append(Badges).Append("\n");
      sb.Append("  ImportantBadges: ").Append(ImportantBadges).Append("\n");
      sb.Append("  IngredientCount: ").Append(IngredientCount).Append("\n");
      sb.Append("  GeneratedText: ").Append(GeneratedText).Append("\n");
      sb.Append("  IngredientList: ").Append(IngredientList).Append("\n");
      sb.Append("  Ingredients: ").Append(Ingredients).Append("\n");
      sb.Append("  Likes: ").Append(Likes).Append("\n");
      sb.Append("  Aisle: ").Append(Aisle).Append("\n");
      sb.Append("  Nutrition: ").Append(Nutrition).Append("\n");
      sb.Append("  Price: ").Append(Price).Append("\n");
      sb.Append("  Servings: ").Append(Servings).Append("\n");
      sb.Append("  SpoonacularScore: ").Append(SpoonacularScore).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/com/spoonacular/client/model/inline_response20030.dart
part of openapi.api;

class InlineResponse20030 {
  
  int id = null;
  
  String title = null;
  
  List<String> breadcrumbs = [];
  
  String imageType = null;
  
  List<String> badges = [];
  
  List<String> importantBadges = [];
  
  int ingredientCount = null;
  
  Object generatedText = null;
  
  String ingredientList = null;
  
  List<InlineResponse20030Ingredients> ingredients = [];
  
  num likes = null;
  
  String aisle = null;
  
  InlineResponse20028Nutrition nutrition = null;
  
  num price = null;
  
  InlineResponse20028Servings servings = null;
  
  num spoonacularScore = null;
  InlineResponse20030();

  @override
  String toString() {
    return 'InlineResponse20030[id=$id, title=$title, breadcrumbs=$breadcrumbs, imageType=$imageType, badges=$badges, importantBadges=$importantBadges, ingredientCount=$ingredientCount, generatedText=$generatedText, ingredientList=$ingredientList, ingredients=$ingredients, likes=$likes, aisle=$aisle, nutrition=$nutrition, price=$price, servings=$servings, spoonacularScore=$spoonacularScore, ]';
  }

  InlineResponse20030.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
    if (json['title'] == null) {
      title = null;
    } else {
          title = json['title'];
    }
    if (json['breadcrumbs'] == null) {
      breadcrumbs = null;
    } else {
      breadcrumbs = (json['breadcrumbs'] as List).cast<String>();
    }
    if (json['imageType'] == null) {
      imageType = null;
    } else {
          imageType = json['imageType'];
    }
    if (json['badges'] == null) {
      badges = null;
    } else {
      badges = (json['badges'] as List).cast<String>();
    }
    if (json['importantBadges'] == null) {
      importantBadges = null;
    } else {
      importantBadges = (json['importantBadges'] as List).cast<String>();
    }
    if (json['ingredientCount'] == null) {
      ingredientCount = null;
    } else {
          ingredientCount = json['ingredientCount'];
    }
    if (json['generatedText'] == null) {
      generatedText = null;
    } else {
      generatedText = new Object.fromJson(json['generatedText']);
    }
    if (json['ingredientList'] == null) {
      ingredientList = null;
    } else {
          ingredientList = json['ingredientList'];
    }
    if (json['ingredients'] == null) {
      ingredients = null;
    } else {
      ingredients = InlineResponse20030Ingredients.listFromJson(json['ingredients']);
    }
    if (json['likes'] == null) {
      likes = null;
    } else {
          likes = json['likes'];
    }
    if (json['aisle'] == null) {
      aisle = null;
    } else {
          aisle = json['aisle'];
    }
    if (json['nutrition'] == null) {
      nutrition = null;
    } else {
      nutrition = new InlineResponse20028Nutrition.fromJson(json['nutrition']);
    }
    if (json['price'] == null) {
      price = null;
    } else {
          price = json['price'];
    }
    if (json['servings'] == null) {
      servings = null;
    } else {
      servings = new InlineResponse20028Servings.fromJson(json['servings']);
    }
    if (json['spoonacularScore'] == null) {
      spoonacularScore = null;
    } else {
          spoonacularScore = json['spoonacularScore'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (id != null)
      json['id'] = id;
    if (title != null)
      json['title'] = title;
    if (breadcrumbs != null)
      json['breadcrumbs'] = breadcrumbs;
    if (imageType != null)
      json['imageType'] = imageType;
    if (badges != null)
      json['badges'] = badges;
    if (importantBadges != null)
      json['importantBadges'] = importantBadges;
    if (ingredientCount != null)
      json['ingredientCount'] = ingredientCount;
    if (generatedText != null)
      json['generatedText'] = generatedText;
    if (ingredientList != null)
      json['ingredientList'] = ingredientList;
    if (ingredients != null)
      json['ingredients'] = ingredients;
    if (likes != null)
      json['likes'] = likes;
    if (aisle != null)
      json['aisle'] = aisle;
    if (nutrition != null)
      json['nutrition'] = nutrition;
    if (price != null)
      json['price'] = price;
    if (servings != null)
      json['servings'] = servings;
    if (spoonacularScore != null)
      json['spoonacularScore'] = spoonacularScore;
    return json;
  }

  static List<InlineResponse20030> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20030>() : json.map((value) => new InlineResponse20030.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20030> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20030>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20030.fromJson(value));
    }
    return map;
  }
}

